File: web/src/types/backtest.ts
export interface BacktestStartRequest {
  trader_id: string
  run_id?: string | null
  start_ts?: number | null
  end_ts?: number | null
  initial_balance?: number | null
  interval?: string | null
}

export interface BacktestRunIdRequest {
  run_id: string
}

export interface BacktestQueryParams {
  run_id?: string | null
  limit?: number | null
}

export interface BacktestRunActionPayload {
  run_id: string
  message: string
}

export interface BacktestMessagePayload {
  message: string
}

export interface BacktestRunSummaryPayload {
  final_equity?: number
  initial_balance?: number
  max_drawdown_pct?: number
  total_trades?: number
  winning_trades?: number
  total_realized_pnl?: number
  max_equity?: number
}

export interface BacktestRunPayload {
  run_id: string
  state: string
  last_error: string
  summary: BacktestRunSummaryPayload
  created_at: number
  updated_at: number
}

export interface BacktestRunsPayload {
  runs: BacktestRunPayload[]
  count: number
}

File: core/src/contracts/backtest.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Deserialize)]
pub struct BacktestStartRequest {
    pub trader_id: String,
    pub run_id: Option<String>,
    pub start_ts: Option<i64>,
    pub end_ts: Option<i64>,
    pub initial_balance: Option<f64>,
    pub interval: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct BacktestRunIdRequest {
    pub run_id: String,
}

#[derive(Debug, Deserialize)]
pub struct BacktestQueryParams {
    pub limit: Option<i64>,
}

#[derive(Debug, Clone, Serialize)]
pub struct BacktestRunActionPayload {
    pub run_id: String,
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct BacktestMessagePayload {
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct BacktestRunsPayload {
    pub runs: Vec<Value>,
    pub count: usize,
}
